test/jdk/java/awt/Window/InvalidFocusLostEventTest/InvalidFocusLostEventTest.java

```diff
@@ -89,10 +95,6 @@ public void actionPerformed(ActionEvent ev) {
 
         // we should check focus after all events are processed,
         // since focus transfers are asynchronous
-        EventQueue.invokeLater(() -> {
-            if (fm.getFocusOwner() != b) {
-                failed = true;
-            }
-        });
+        EventQueue.invokeLater(() -> countDownLatch.countDown());
```

Comment:
I guess we can just do countDown without it being in invokeLater...

Reply:
I have included countDown specifically inside asynchronous call of EDT because of the comment mentioned above `we should check focus after all events are processed, since focus transfers are asynchronous`.

So that we actually check for focus after everything is processed in EDT.

Reply:
But I am not sure this is the correct way to ensure...We are trying to do away with invokeLater call from jtreg tests.
I think it's more apt to call robot.waitForIdle to ensure all events are processed
void
[waitForIdle](https://docs.oracle.com/en/java/javase/22/docs/api/java.desktop/java/awt/Robot.html#waitForIdle())()
Waits until all events currently on the event queue have been processed.

Reply:
Adding waitForIdle() in EDT(actionPerformed()) is throwing IllegealThreadStateException().

So i refactored code to remove the usage of CoutnDownLatch itself and using waitForIdle() in main(). This will serve our purpose to wait until all events are processed and then we can check for current focused UI component.